src/varname.jl

```diff
@@ -26,29 +29,42 @@ julia> @varname x[:, 1][1+1]
 x[:,1][2]
 ```
 """
-struct VarName{sym, T<:Tuple}
+struct VarName{sym, T<:Lens}
     indexing::T
 
-    VarName{sym}(indexing::Tuple=()) where {sym} = new{sym,typeof(indexing)}(indexing)
+    VarName{sym}(indexing=IdentityLens()) where {sym} = new{sym,typeof(indexing)}(indexing)
 end
 
+# A bit of backwards compatibility.
+# TODO: Should we deprecate this?
```

Comment:
I'm not unhappy with keeping around that variant.

Reply:
But forever or temporarily? If forever (it's honestly a convenient constructor for `VarName` dealing with arrays only), then there's no need to deprecate. If temporarily we should deprecate.

Reply:
I'm fine with either.

Reply:
> it's honestly a convenient constructor for `VarName` dealing with arrays only

Exactly my reasoning.